src/api/catalog.ts

```diff
@@ -83,99 +82,98 @@ export default ({config, db}) => async function (req, res, body) {
     }
   }
 
-  // pass the request to elasticsearch
-  const elasticBackendUrl = adjustBackendProxyUrl(req, indexName, entityType, config)
```

Comment:
In `adjustBackendProxyUrl` there is some logic that modify url. For example `_source_include` and `_source_exclude`. Can you add it somewhere?

Reply:
I see that you added `adjustQueryParams` but we are missing `adjustQueryParams` and 
```
    if (config.elasticsearch.cacheRequest) {
      parsedQuery.request_cache = !!config.elasticsearch.cacheRequest
    }
```
I think that will be all :thinking:

Reply:
I've rearranged it a bit, so it could be used  again but uses the same logic of `adjustQueryParams`. I also stripped down the `if/else` a bit as it used redundant commands.

Reply:
And yes, I used another method because the URL is already handled by `getClient` and the index name by `adjustQuery`.

Reply:
ok cool :) One more thing https://github.com/DivanteLtd/vue-storefront-api/pull/512/files#diff-5d56871e4ca9c6f57eeb1e427bc46dc9R82 here is function `decorateBackendUrl` which holds some logic. We need to keep it also

Reply:
Oh, alright, sorry. Do you mean the `useRequestFilter` logic? As we don't put those information into the request-url and in the clients search-config params instead I'd also add this logic to `adjustQueryParams`, wdyt? And do I see it right that `decorateBackendUrl` is also missing ES7 support as it only uses `_source_include` instead of `_source_includes`.

Reply:
Okay, I've made an update and added the `useRequestFilter` and `overwriteRequestSourceParams` logic to `adjustQueryParams`. I also added ES7 support to `decorateBackendUrl` by using the correct `_source_exclude`/`_source_include` params depending on the ES version.

Reply:
looks good for me 👍